elide-core/src/main/java/com/yahoo/elide/core/PersistentResource.java

```diff
@@ -483,20 +483,28 @@ public boolean clearRelation(String relationName) {
     public void removeRelation(String fieldName, PersistentResource removeResource) {
         checkPermission(UpdatePermission.class, this);
         checkFieldPermission(UpdatePermission.class, this, fieldName);
+        checkPermission(DeletePermission.class, removeResource);
```

Comment:
I didn't change the permissions.  I just eliminated duplicates and stuck them in new places.